### ISSUE TITLE
Blazor WebAssembly & Azure.Storage.Blobs upload freezing/hanging

### ISSUE THREAD

**Issue by daltskin** [dotnet/aspnetcore#18852]
### Describe the bug
When using the Azure.Storage.Blobs package from Blazor wasm to upload files to blob storage the application is freezing, even though the upload succeeds to the blob storage endpoint (files can be then be retrieved to ensure they've been written correctly).

There is a difference in behaviour if the result of the UploadAsync is awaited or not (see code extract below).  If it's awaited, the first file in the files collection will be successfully written to blob storage - but then the UI will freeze.  If the result is not awaited all the files in the collection will be successfully written to blob storage, then the UI will freeze.

### To Reproduce

What steps can we follow to reproduce the issue?

Using Azure.Storage.Blobs nuget package (12.2.0)

``` C#
BlobServiceClient service = new BlobServiceClient(<blob_connection_string>);
BlobContainerClient container = service.GetBlobContainerClient(<blob_container>);
foreach (var file in _files)
{
   var blobClient = container.GetBlobClient(file.Name);
   // get file stream (ms)...
   
   // This will successfully upload the file to blob storage, but then hang on the first file
   await blobClient.UploadAsync(ms, overwrite: true);

   // If you don't await the result like the below-  all files will will successfully upload to blob storage, then it will hang
   blobClient.UploadAsync(ms, overwrite: true, System.Threading.CancellationToken.None);
}
```
Full sample here: https://github.com/daltskin/BlazorFile2Azure/blob/blobsdk/Client/Pages/UploadFileDirect.razor

WASM debugging exception:
Exception: ErrnoError
error: 44
message: "FS error"
node: undefined
name: "Azure.Storage.Blobs.dll"
![blazorblob](https://user-images.githubusercontent.com/1678318/73937974-2b9efa00-48de-11ea-9fab-3499fcdabe15.png)

### Further technical details
- ASP.NET Core version: 3.1.200-preview-014883
-- Blazor v3.2.0-preview1.20073.1
- Include the output of `dotnet --info`
 OS Name:     Windows
 OS Version:  10.0.18363
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\3.1.200-preview-014883\

- Visual Studio 2019 Preview 16.5.0 Preview 2.0

**Comment by javiercn** [comment]
@daltskin thanks for contacting us.

I believe this is likely caused by the library internally doing .Wait() or similar. That normally doesn't cause issues as in most cases there is a separate thread that can resume the execution and unblock the waiting thread.

In the Blazor case, there is only a single thread, so if the library blocks internally, or does something that is not totally correct with the thread management, then it is likely to block the UI completely and given that Blazor is single threaded, there's no other thread to notify the blocked thread and resume the execution.

@danroth27 this is external, can you follow up with someone on the SDK team to get to the root of this?

**Comment by mkArtakMSFT** [comment]
@pakrym do you know whether there are blocking calls in this package? Can you recommend where we should move this issue to?

**Comment by pakrym** [comment]
The `*Async` overloads should be purely async all the way to HttpClient.

**Comment by mkArtak** [comment]
I tried this out, and it's not freezing / hanging for me. Instead, I get `System.Threading.SynchronizationLockException: Cannot wait on monitors on this runtime.` exception. Here is the full stack trace:
![image](https://user-images.githubusercontent.com/7347708/80427043-cd116880-889b-11ea-874e-bd1f7f75de9c.png)


Seems this is the underlying exception:
![image](https://user-images.githubusercontent.com/7347708/80427108-ea463700-889b-11ea-90ed-e436242e0fa2.png)


This is the code I used to simulate this:
```csharp
BlobClient client = new BlobClient("***", "***", "bla");
return client.UploadAsync(payload);
```

And I'm referencing the following package: 
```
<PackageReference Include="Azure.Storage.Blobs" Version="12.4.1" />
```

**Comment by pranavkm** [comment]
https://github.com/Azure/azure-sdk-for-net/blob/29fd2754c3830a43feaa04784030a37bc6121621/sdk/core/Azure.Core/src/Pipeline/HttpClientTransport.cs#L478

**Comment by mkArtakMSFT** [comment]
I'm going to move this issue into the `Azure/azure-sdk-for-net` repository as the fix is on their side.

**Comment by mkArtakMSFT** [comment]
Ok, filed a separate issue in the Azure SDK repo to track this: https://github.com/Azure/azure-sdk-for-net/issues/11626
